src/app/zap-templates/zcl/data-model/chip/mode-select-cluster.xml

```diff
@@ -17,9 +17,30 @@ limitations under the License.
 <configurator>
   <domain name="CHIP"/>
 
+  <enum name="ChangeToModeResponseStatus" type="ENUM8">
+    <cluster code="0x0050"/>
+    <item value="0x0" name="Success"/>
+    <item value="0x1" name="UnsupportedMode"/>
+    <item value="0x2" name="GenericFailure"/>
+  </enum>
+
+  <enum name="SemanticTags" type="ENUM16">
```

Comment:
This is not actually an enum in the spec, right?  Or should it be treated as one?  This is going to be generated as an enum class, which means values other than these will not really be usable in a "semantic tag" as defined by this enum.  Luckily we are not using this enum type in SemanticTagStruct, but then the question is: why have this in the XML at all?

If we do keep it in the XML, we need to sort out the naming.  This is not the set of all semantic tags; this is the "common standard semanic tags", and the enum name should probably reflect that.  But maybe the fact that this is scoped to this cluster already does?  This is where the fact that this is not an enum in the spec and hence does not have a defined name is a problem.

Reply:
It is not explicitly mentioned in the spec that these should be enums. The reason why I thought they should be in the XML was that it would be helpful to the consumer of the SDK. 

As you pointed out, the `Value` item in the `SemanticTagStruct` is of type `ENUM16` which means that enums from the different namespaces can be used by casting them to `uint16_t`. I think that this save the consumer from having to define these values themselves and ensures that clients and servers are using the same values.

My understanding is that the `SemanticTags` enum in the `ModeSelect` namespace is generic just like the cluster itself.

Happy to remove them and allow the user to look up the spec to set these values though if we think that this is bloating the SDK.

Reply:
I believe that we should a new abstraction for those tags, to avoid generating insane amounts of duplicated elements or overlapping structs in a way that's it's not clear that there is overlap.

Reply:
@tcarmelveilleux I'm not quite following the concern here.  For each cluster, SemanticTags are namespaced to that cluster, no?

Reply:
Yes they are for the aliased clusters. Only the tags defined in the `ModeSelect` namespace can be used by other clusters.